client/signup/main.jsx

```diff
@@ -192,17 +192,19 @@ class Signup extends React.Component {
 		// a WP.com account during the signup flow.
 		this.completeP2FlowAfterLoggingIn();
 
-		if ( canResumeFlow( this.props.flowName, this.props.progress ) ) {
+		if ( canResumeFlow( this.props.flowName, this.props.progress, this.props.isLoggedIn ) ) {
 			// Resume from the current window location
 			return;
 		}
 
 		if ( this.getPositionInFlow() !== 0 ) {
 			// Flow is not resumable; redirect to the beginning of the flow.
 			// Set `resumingStep` to prevent flash of incorrect step before the redirect.
-			const destinationStep = flows.getFlow( this.props.flowName ).steps[ 0 ];
+			const destinationStep = flows.getFlow( this.props.flowName, this.props.isLoggedIn )
+				.steps[ 0 ];
 			this.setState( { resumingStep: destinationStep } );
-			return page.redirect( getStepUrl( this.props.flowName, destinationStep, this.props.locale ) );
```

Comment:
Was there a bug here, passing `locale` as the `stepSectionName` parameter?

Reply:
Yep 👍